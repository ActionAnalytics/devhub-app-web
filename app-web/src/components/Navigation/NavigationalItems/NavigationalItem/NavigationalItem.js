import React from 'react';
import { push } from 'gatsby-link';
import PropTypes from 'prop-types';
import HexBlock from '../../../UI/HexBlock/HexBlock';
import classes from './NavigationalItem.module.css';

const NavigationalItem = props => (
  <HexBlock
    icon={props.icon}
    link={props.link}
    collapses
    gridClassNumber={props.hexGridClassNumber}
  >
    <span className={classes.NavigationalItem}>
      <span className={classes.Title}>{props.title}</span>
      {props.description && (
        <span className={classes.Description}>{props.description}</span>
      )}
    </span>
  </HexBlock>
);

NavigationalItem.propTypes = {
  icon: PropTypes.string,
  title: PropTypes.string.isRequired,
  description: PropTypes.string,
  link: PropTypes.string.isRequired,
  hexGridClassNumber: PropTypes.number.isRequired,
};

NavigationalItem.defaultProps = {
  icon: '',
  description: '',
};

export default NavigationalItem;
