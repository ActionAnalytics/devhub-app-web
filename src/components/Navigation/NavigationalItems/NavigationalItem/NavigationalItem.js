import React from 'react'
import PropTypes from 'prop-types';
import classes from './NavigationalItem.module.css';
import hexImage from '../../../../assets/images/hex#.svg';

const NavigationalItem = ({ children, bgColor }) => {
    return (
        <li className={classes.NavigationalItem}>
            <div>
                <img src={ "https://farm3.staticflickr.com/2827/10384422264_d9c7299146.jpg" } alt="hex background" />
                { children }
            </div>
        </li>
    )
}

export default NavigationalItem;

