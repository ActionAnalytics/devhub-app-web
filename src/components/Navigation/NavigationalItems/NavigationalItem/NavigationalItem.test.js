import React from 'react';
import { shallow } from 'enzyme';
import NavigationalItem, { navigate } from './NavigationalItem';

describe('Navigational Item Component', () => {
  test('it matches snapshot', () => {
    const baseProps = {
      title: 'this is the title',
      description: 'this is the description',
      link: '/',
    };

    const wrapper = shallow(<NavigationalItem {...baseProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});
