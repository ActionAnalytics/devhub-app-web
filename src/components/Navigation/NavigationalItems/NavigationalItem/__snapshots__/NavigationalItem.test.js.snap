// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigational Item Component it matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NavigationalItem
    description="this is the description"
    icon=""
    link="/"
    title="this is the title"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <div>
        <h3
          className="Title"
        >
          this is the title
        </h3>
        <p>
          props.description
        </p>
      </div>,
      "clicked": [Function],
      "collapses": true,
      "icon": "",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3
            className="Title"
          >
            this is the title
          </h3>,
          <p>
            props.description
          </p>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "this is the title",
            "className": "Title",
          },
          "ref": null,
          "rendered": "this is the title",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "props.description",
          },
          "ref": null,
          "rendered": "props.description",
          "type": "p",
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <div>
          <h3
            className="Title"
          >
            this is the title
          </h3>
          <p>
            props.description
          </p>
        </div>,
        "clicked": [Function],
        "collapses": true,
        "icon": "",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3
              className="Title"
            >
              this is the title
            </h3>,
            <p>
              props.description
            </p>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "this is the title",
              "className": "Title",
            },
            "ref": null,
            "rendered": "this is the title",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "props.description",
            },
            "ref": null,
            "rendered": "props.description",
            "type": "p",
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
